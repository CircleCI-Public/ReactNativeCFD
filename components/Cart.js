import CartItem from './CartItem';
import * as React from 'react';
import {View} from 'react-native';
import MenuData from '../data/MenuData';
import {useEffect, useState} from 'react';
import {Text} from 'react-native-paper';

const Cart = ({cart, removeFromCart}) => {
  if (cart !== undefined) {
    return (
      <View style={{paddingBottom: 60}}>
        {cart.map((menuItem, i) => (
          <>
            <CartItem
              key={i}
              index={i - 1}
              menuItem={menuItem}
              removeFromCart={removeFromCart}
            />
          </>
        ))}
      </View>
    );
  } else {
    return (
      <View style={{paddingBottom: 60}}>
        <Text>Your Cart is Empty</Text>
      </View>
    );
  }
};
export default Cart;
