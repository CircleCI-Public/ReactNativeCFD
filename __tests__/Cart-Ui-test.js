import React from 'react';
import renderer from 'react-test-renderer';
import Cart from '../components/Cart';

test('renders correctly', () => {
  const tree = renderer.create(<Cart />).toJSON();
  expect(tree).toMatchSnapshot();
});
