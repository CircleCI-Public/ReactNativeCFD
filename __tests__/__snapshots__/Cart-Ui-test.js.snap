// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "paddingBottom": 60,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    Your Cart is Empty
  </Text>
</View>
`;
